### PR TITLE
Add registration endpoint

### PR DESCRIPTION
**Pull request by gaurav-handysolver**
**Add registration endpoint.**
* Now user can register and be added in default group without any auth bearer token.

Working Flow:

1. Firstly, user enter their basic details like username, password, confirmPassword, firstname(optional), lastname(optional) etc.
 * We also provide pre-filled user data.
 * It will also check password and confirmPassword are equal, if not then it will show a message. 
 
2. When you hit the API url, it will add user details in user table and then the user will automatically add in "USER" group (this is the humhub default group which has ID=2).  

3. And Finally, username and password will be used to authenticate the user with userlogin method. if the username and password will be correct then user will get success message.

Points to remember before using the API
1. Module should be activated(checked) from the core rest module configuration setting, otherwise it will thrown an error (Invalid route).
2. And all users/specific users should be allowed to use API
image

#2

### CODE REVIEW

**Comment by donni106** [comment]
I change the target branch to issue-1 to not include the changes from that PR here. Once #11 is merged, this will be changed to `main` by GitHub automatically.

File: controllers/user/GroupController.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const USER_DEFAULT_GROUP_ID = 2;
+
+    /**
+     * Adds the user to a particular humhub group by passing groupID and userID
+     *
+     * @param $id
+     * @param $userId
+     * @return array
+     */
+    public function actionMemberAddWithoutAuth($id,$userId)
```
Comment (donni106) [inline controllers/user/GroupController.php:25]: isn't this another topic and not user registration? or am i wrong?

File: controllers/user/UserController.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+     * Register user without bearer token
+     *
+     * Steps
+     * 1. First, The user register in 'user' table
+     * 2. New user will be automatically adds in default humhub group, which is 'USER' Group whose (id=2)
+     * 3. And finally, User will login with credentials(username and password)
+     *
+     * @return array
+     */
+    public function actionCreate()
```
Comment (donni106) [inline controllers/user/UserController.php:29]: what is the difference between the use controller in `auth/user` and `user`? they seem very similar to me.

File: controllers/user/GroupController.php (hunk 1 above)
Comment (gaurav-handysolver) [reply controllers/user/GroupController.php:25]: Hi @donni106 
> New users should be added to the HumHub "Default Group" automatically.

As you mentioned your requirement in issue 2. This controller is used to add registered user in humhub default group which is 'USER' Group whose has id=2.
image

File: controllers/user/UserController.php (hunk 2 above)
Comment (gaurav-handysolver) [reply controllers/user/UserController.php:29]: Hi @donni106 
As far as we know, there is only one usercontroller in the module. Earlier usercontroller was in auth folder but in our last commit we removed the auth folder in controllers and refactor the namespaces.

Or you are pointing out something else?
Thanks!

Comment (donni106) [reply controllers/user/UserController.php:29]: i meant the duplication of files which you can see in the file structure:

Bildschirmfoto 2022-04-29 um 14 35 58

Comment (gaurav-handysolver) [reply controllers/user/UserController.php:29]: Hi @donni106 
Thanks for pointing out. Now the duplication of files are gone on the latest commit we did.

**Review by donni106 — APPROVED** [review]
Nice! Module successfully installed!

|1|2|3|
|---|---|---|
|Bildschirmfoto 2022-05-02 um 11 37 28|Bildschirmfoto 2022-05-02 um 11 38 01|Bildschirmfoto 2022-05-02 um 11 38 10|

Tests also successful:

|params missing|params already used|params existing|
|---|---|---|
|Bildschirmfoto 2022-05-02 um 11 41 04|Bildschirmfoto 2022-05-02 um 11 46 41|Bildschirmfoto 2022-05-02 um 11 45 20|

**Pull request merged**